File: Prog_Skill/truck.py
arr = []
n = int(input("Enter the number of elements in the list: "))
w = int(input("Enter the weight of the truck: "))
while 1 <= n and w <= 10000:
    while n != 0 and w != 0:
        for i in range(n):
            arr.append(int(input()))
        temp = 0
        truckCount = 0
        for i in range(n):
            if temp < arr[i]:
                temp = w
                truckCount += 1
            temp -= arr[i]

        print(truckCount)
        n = int(input("Enter the number of elements in the list: "))
        w = int(input("Enter the weight of the truck: "))

File: Prog_Skill/truck.cpp
#include <iostream>
using namespace std;

int main()
{
    int size, arr[0],n, w, temp = 0, truckCount[2] = {0,0} ,tc_size = -1;
    while (n != 0 && w != 0)
    {
        cin >> n;
        cin >> w;
        size = n;
        arr[size];
        tc_size++;
        for (int i = 0; i < n; i++)
        {
            cin >> arr[i]; //input the array
        }
        for (int i = 0; i < n; i++)
        {
            if (temp < arr[i])
            {
                temp = w;
                truckCount[tc_size]++;
            }
            temp -= arr[i];
        }
    }
    cout << truckCount[0] << "\n" << truckCount[1];

    return 0;
}